recipe-server/client/control_new/components/common/ShieldIdenticon.js

```diff
@@ -146,88 +145,120 @@ class Cover extends React.Component {
   }
 }
 
+
+function hexLuma(hex) {
+  // Quick HEX to RGB.
+  const hexRgb = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+  const R = parseInt(hexRgb[1], 16);
+  const G = parseInt(hexRgb[2], 16);
+  const B = parseInt(hexRgb[3], 16);
+
+  // Determine color luminance based on perceived values.
+  return (0.2126 * R) + (0.7152 * G) + (0.0722 * B);
+}
+
+function pickPair(genome, base) {
+  const baseLuminance = hexLuma(base);
+  let chosenColor;
+  let pairLuminance;
+
+  do {
+    chosenColor = genome.choice(colors, 'counter color');
+    pairLuminance = hexLuma(chosenColor);
+  } while (Math.abs(baseLuminance - pairLuminance) < 75);
```

Comment:
This works well, but I worry about it running a variable number of times, each time using up a bunch of entropy from the `Genome`. Could we instead filter `colors` to remove colors that aren't different enough in luminescence, and then run `genome.choice` only once?